pkg/provider/azure/data/spot.go

```diff
@@ -6,392 +6,436 @@ import (
+	for l, pss := range args.spotPricings {
+		validImage := true
+		if args.imageRef != nil {
+			validImage = IsImageOffered(
+				ImageRequest{
+					Region:         l,
+					ImageReference: *args.imageRef,
+				})
+			if err != nil {
+				return nil, err
+			}
+		}
+		for _, ps := range pss {
+			idx := slices.IndexFunc(args.evictionRates[l],
+				func(evr evictionRateResult) bool {
+					return evr.Location == ps.Location && validImage
+				})
+			if idx != -1 {
+				result[l] = &SpotInfoResult{
+					Location:     args.evictionRates[l][idx].Location,
+					Price:        ps.Price,
+					ComputeSize:  ps.ComputeSize,
+					EvictionRate: args.evictionRates[l][idx].EvictionRate,
+				}
+			}
+		}
 	}
-	i := slices.IndexFunc(
-		ers,
-		func(e evictionRateSpec) bool {
-			return e.id == er
+	spis := utilMaps.Values(result)
+	utilSlices.SortbyFloat(spis,
+		func(s *SpotInfoResult) float64 {
+			return s.Price
 		})
-	return ers[i].value
-}
-
-// Get eviction rate parsing its name
-func parseEvictionRate(str string) (EvictionRate, bool) {
-	c, ok := evictionRates[strings.ToLower(str)]
-	return c.id, ok
+	if len(spis) == 0 {
+		return nil, fmt.Errorf("no good choice was found")
+	}
+	return spis[0], nil
 }
 
-// This is a fallback function in case we need to get an option only based in price
-// In order to add some type of distribution across the information we will 1/3 at beguining
-// 1/3 at the end and then randomly we will pick one of the remaining
-func getSpotChoiceByPrice(s []priceHistory, imageID string) (*BestSpotChoiceResponse, error) {
-	var spotChoices []*BestSpotChoiceResponse
-	for _, sv := range s {
-		ir := ImageRequest{
-			Region: sv.Location,
-			ImageReference: ImageReference{
-				ID: imageID,
-			},
-		}
-		if IsImageOffered(ir) {
-			spotChoices = append(spotChoices,
-				&BestSpotChoiceResponse{
-					VMType:   sv.VMType,
-					Location: sv.Location,
-					Price:    sv.Price,
-				})
-
-		}
+// // This is a fallback function in case we need to get an option only based in price
+// // In order to add some type of distribution across the information we will 1/3 at beguining
+// // 1/3 at the end and then randomly we will pick one of the remaining
+// func getSpotChoiceByPrice(s []priceHistory, imageID string) (*BestSpotChoiceResponse, error) {
+// 	var spotChoices []*BestSpotChoiceResponse
+// 	for _, sv := range s {
+// 		ir := ImageRequest{
+// 			Region: sv.Location,
+// 			ImageReference: ImageReference{
+// 				ID: imageID,
+// 			},
+// 		}
+// 		if IsImageOffered(ir) {
+// 			spotChoices = append(spotChoices,
+// 				&BestSpotChoiceResponse{
+// 					VMType:   sv.VMType,
+// 					Location: sv.Location,
+// 					Price:    sv.Price,
+// 				})
+
+// 		}
+// 	}
+// 	if len(spotChoices) > 3 {
+// 		return util.RandomItemFromArray(
+// 				spotChoices[len(spotChoices)/3 : len(spotChoices)-len(spotChoices)/3]),
+// 			nil
+// 	}
+// 	return util.RandomItemFromArray(spotChoices), nil
+// }
+
+// TODO review this one
```

Comment:
do you mean the `getSpotChoiceByPrice` or the `osType` function? if the later maybe remove the commented code for `getSpotChoiceByPrice`

Reply:
oh yeah that is a leftover I need to take that into consideration when the evition rate query fails...almost forgot

the TODO review is for osType (to just clean it up a bit)

Reply:
Fixed